src/layout/barPolar.js

```diff
@@ -83,6 +83,19 @@ function barLayoutPolar(seriesType, ecModel, api) {
             || !seriesModel.get('roundCap', true);
 
         var valueAxisStart = valueAxis.getExtent()[0];
+        var radiusValueAxisStart;
+        if (valueAxis.dim === 'radius') {
+            radiusValueAxisStart = valueAxis.model.option.min || valueAxis.getExtent()[0];
+            if (valueAxis.model.option.min && valueAxis.model.option.min < 0) {
+                if (!valueAxis.model.option.max
+                    || valueAxis.model.option.max && valueAxis.model.option.max > 0) {
+                    radiusValueAxisStart = valueAxis.getExtent()[0];
+                } else if (valueAxis.model.option.max && valueAxis.model.option.max < 0) {
+                    radiusValueAxisStart = valueAxis.model.option.max;
+                }
+            }
+        }
+
```

Comment:
`valueAxisStart` means the `0` position in the axis extent. You don't need to do all these logics to calculate the `max` because it's already in `valueAxis.scale.getExtent()`.

So my suggestion for this part would be only:
```js
var valueAxisStart = valueAxis.dim === 'radius' 
    ? valueAxis.dataToRadius(0) 
    : valueAxis.dataToAngle(0);
```